server/apps/immich/src/modules/schedule-tasks/schedule-tasks.service.ts

```diff
@@ -1,26 +1,12 @@
-import { Inject, Injectable } from '@nestjs/common';
+import { JobName, JobService } from '@app/domain';
+import { Injectable } from '@nestjs/common';
 import { Cron, CronExpression } from '@nestjs/schedule';
-import { InjectRepository } from '@nestjs/typeorm';
-import { IsNull, Not, Repository } from 'typeorm';
-import { UserEntity } from '@app/infra';
-import { userUtils } from '@app/common';
-import { IJobRepository, JobName } from '@app/domain';
 
 @Injectable()
 export class ScheduleTasksService {
-  constructor(
-    @InjectRepository(UserEntity)
-    private userRepository: Repository<UserEntity>,
-
-    @Inject(IJobRepository) private jobRepository: IJobRepository,
-  ) {}
+  constructor(private jobService: JobService) {}
   @Cron(CronExpression.EVERY_DAY_AT_11PM)
-  async deleteUserAndRelatedAssets() {
-    const usersToDelete = await this.userRepository.find({ withDeleted: true, where: { deletedAt: Not(IsNull()) } });
-    for (const user of usersToDelete) {
-      if (userUtils.isReadyForDeletion(user)) {
-        await this.jobRepository.add({ name: JobName.USER_DELETION, data: { user } });
-      }
-    }
+  async onUserDeleteCheck() {
+    await this.jobService.handle({ name: JobName.USER_DELETE_CHECK });
```

Comment:
My mental mind set is still not completely registered with this, I was expecting the method `handle` to be found in `api-v1/job/job.service.ts` but it is in the `domain`. What should I do to better mapping the flow in my head?

Reply:
Everything in Job service in the API will be merged into the domain, just that code has a lot of dependency on after repository and that isn't in the domain yet, so it is a work in progress.